Makefile

```diff
@@ -22,17 +22,7 @@ include Makefile.common
 
 # For users who don't read the INSTALL file
 .PHONY: defaultentry
-defaultentry:
-ifeq "$(UNIX_OR_WIN32)" "unix"
-	@echo "Please refer to the installation instructions in file INSTALL."
-	@echo "If you've just unpacked the distribution, something like"
-	@echo "	./configure"
-	@echo "	make world.opt"
-	@echo "	make install"
-	@echo "should work.  But see the file INSTALL for more details."
-else
-	@echo "Please refer to the instructions in file README.win32.adoc."
-endif
+defaultentry: $(DEFAULT_MAKE_TARGET)
```

Comment:
The [GNU Coding Standards](https://www.gnu.org/prep/standards/html_node/Standard-Targets.html) recommend this target be called `all`.

At the moment, the `all` target rebuilds the system using whatever is presently in `boot/` (so it's used by `world` after first building the runtime and `bootstrap` after a bootstrap cycle). It seems to me we could safely rename that `all` target and use it here?

Reply:
Let me explain the current Makefile, because it's better to understand it before changing it.  

The entries are named from the perspective of a developer (someone who changes the OCaml sources) rather than the perspective of a user (someone who just builds from a source distribution).

`world` is the initial build from a fresh git checkout.  It populates `boot/` with a bytecode interpreter and a standard library, then rebuilds everything (in bytecode at least).

After this initialization, `all` is the "normal" build for a developer: after a change, it rebuilds everything in bytecode, using the existing `boot/`.  For many changes -- all those that require a bootstrap later -- it would be wrong to do `world` and rebuild `boot/`.  This could leave the working copy in an inconsistent state, requiring the changes to be rolled back.

The subtlety above is lost on many core developers, who always blindly do `make world`, then complain that bootstrapping is difficult.  So, maybe changing the meaning of `make all` entirely would just add to the existing confusion in negligible amounts.  Still, there is confusion that needs to be taken into account.  "The GNU Coding Standards recommend this and  that" is not enough.

Reply:
Indeed, but I guess I'm asking for `world` to become `all` and `all` to become something else (like `system`, or `tools`, or `compiler` or something else which doesn't imply *everything*)

Reply:
>  I'm asking for world to become all

I understood you're asking for `defaultentry` to become `all`.  (`defaultentry` being `world` or `world.opt` depending on platform.)

My point is that `all` (today) has a well defined meaning for core system developers.  There is also `allopt` that is like `all` but for the native-code parts of the system.  We need to come up with a better story than just rename `defaultentry` to `all` and `all` to "`system`, or `tools`, or `compiler` or something else".

Reply:
Yes, sorry - wrote too quickly as to which target was being renamed. With the aid of a fairly trivial "make cookie" (just `boot/built` or something) could we have it that `make all` the first time does `world` (or `world.opt`) as appropriate but the second time (because of the cookie) does exactly what `make all` does now? `make world` continues to mean the same as it did before, but most of the time it means that just `make` on its own actually does the correct thing, both for a core developer and for an end-user?